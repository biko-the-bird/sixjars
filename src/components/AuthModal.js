import React, { Component } from "react";

import Gun from 'gun';
require('gun/sea');
const gun = Gun();
const user = gun.user();

export class AuthModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: ""
    };
  }

  signup = () => {
    user.create(
      this.state.username,
      this.state.password,
      res => {
        console.log("user created", res);
        console.log("sea2", user._.sea);
        user.recall();
        console.log("sea3", user._.sea);
      },
      []
    );
  };
  login = () => {
    user.create(
      this.state.username,
      this.state.password,
      res => {
        console.log("user created", res);
        console.log("sea2", user._.sea);
        user.leave();
        console.log("sea3", user._.sea);
      },
      []
    );
  };

  validInputs = () => {
    var password = this.state.password;
    var username = this.state.username;
    var errors = [];
    if (password.length < 6) {
      console.log(password.length);
      errors.push("password must be >6 characters");
    }
    if (username === "") {
      errors.push("a username is required");
    }

      
    return (
      <div>
      {(errors.length > 0) 
        ?(
          <ul>
          {errors.map(err => 
            <li key={err}><span className="ghost">{err}</span></li>
          )}
          </ul>
        )
        :(
          <div>
          <button className="ghost-button " onClick={() => this.signup()}>Signup</button>
     
          <button className="ghost-button " onClick={() => this.login()}>Login</button>
          </div>
        )
      
      }
      </div>
    )
     
     
      

      
    
  };

  

  render() {
    return (
      <div className="auth-modal">
        <h1 className="ghost"><u>Sync Account</u></h1>
        <p className="ghost">Email</p>
        <input 
        type="text"
        placeholder="email"
        className="ghost-input username-input" 
        onChange={e => this.setState({username: e.target.value})}
        />
        <br/>

        <p className="ghost">Password</p>
        <input 
        type="password"
        placeholder="password"
        className="ghost-input password-input" 
        onChange={e => this.setState({password: e.target.value})}
        />
        {this.validInputs()}
  
        <div />
      </div>
    );
  }
}
export default AuthModal;
