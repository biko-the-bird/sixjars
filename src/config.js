
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//this is a configuration for enzyme that needs
//to be imported in each testing file
const config = configure({ adapter: new Adapter() });
export default config;