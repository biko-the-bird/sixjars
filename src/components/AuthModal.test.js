import React from 'react';
import {shallow} from 'enzyme';
import config from '../config';
import AuthModal from './AuthModal'

describe("authModal", () => {
    const authmodal = shallow(<AuthModal/>);
    const expectedUsername = "janedoe";
    const expectedPass = "password";
    it('renders', () => {
        expect(authmodal).toMatchSnapshot();
    });
    describe(('username input'),() => {
        it('contains a input for username', () => {
            expect(authmodal.find('.username-input').exists()).toBe(true);
        });
        describe('when user enters a `username` input', () => {
            
            beforeAll(() => {
                authmodal.find('.username-input').simulate('change', {target: {value: expectedUsername}})
            });
            it("updates the local `state`", () => {
                expect(authmodal.state().username).toEqual(expectedUsername);
            });
        });
        describe('when user enters a `password` input', () => {
            
            beforeAll(() => {
                authmodal.find('.password-input').simulate('change', {target: {value: expectedPass}})
            });
            it("updates the local `state`", () => {
                expect(authmodal.state().password).toEqual(expectedPass);
            });
        });
        describe('when user enters a `password` input', () => {
            const expectedPass = "password";
            beforeAll(() => {
                authmodal.find('.password-input').simulate('change', {target: {value: expectedPass}})
            });
            it("updates the local `state`", () => {
                expect(authmodal.state().password).toEqual(expectedPass);
            });
        });
        describe('only show `login` and `signup` buttons when there is valid input', () => {
            it('contains a ul when there are no inputs', () => {
                expect(authmodal.find('ul').exists()).toBe(true);
            });
            describe ("when there are valid inputs", () => {
                beforeAll(() => {
                    authmodal.find('.username-input').simulate('change', {target: {value: expectedUsername}})
                    authmodal.find('.password-input').simulate('change', {target: {value: expectedPass}})
                });
                it("renders buttons", () => {
                    expect(authmodal.find('button').exists()).toBe(true);
                });
            })
           
        });
    })
   
})