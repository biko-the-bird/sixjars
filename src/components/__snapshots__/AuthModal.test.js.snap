// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authModal renders 1`] = `ShallowWrapper {}`;
